OrientationAnalysis/src/OrientationAnalysis/Filters/ConvertOrientations.cpp

```diff
@@ -302,253 +446,351 @@ Result<> ConvertOrientations::executeImpl(DataStructure& dataStructure, const Ar
   }
   else if(inputType == OrientationRepresentation::Type::Cubochoric && outputType == OrientationRepresentation::Type::Quaternion)
   {
     messageHandler(complex::IFilter::Message{complex::IFilter::Message::Type::Info, "Converting Cubochoric to Quaternion"});
     ToQuaternionFunctionType cu2qu = OrientationTransformation::cu2qu<InputType, QuaternionType>;
-    parallelAlgorithm.execute(::ToQuaternion<float, ToQuaternionFunctionType, 3, 4>(inputDataArray, outputDataArray, cu2qu, QuaternionType::Order::VectorScalar));
+    parallelAlgorithm.execute(
+        ::ToQuaternion<float, ToQuaternionFunctionType, ValidateInputDataFunctionType, 3, 4>(inputDataArray, outputDataArray, cu2qu, cuCheck, QuaternionType::Order::VectorScalar));
   }
   else if(inputType == OrientationRepresentation::Type::Cubochoric && outputType == OrientationRepresentation::Type::AxisAngle)
   {
     messageHandler(complex::IFilter::Message{complex::IFilter::Message::Type::Info, "Converting Cubochoric to AxisAngle"});
     ConversionFunctionType cu2ax = OrientationTransformation::cu2ax<InputType, OutputType>;
-    parallelAlgorithm.execute(::ConvertOrientation<float, ConversionFunctionType, 3, 4>(inputDataArray, outputDataArray, cu2ax));
+    parallelAlgorithm.execute(::ConvertOrientation<float, ConversionFunctionType, ValidateInputDataFunctionType, 3, 4>(inputDataArray, outputDataArray, cu2ax, cuCheck));
   }
   else if(inputType == OrientationRepresentation::Type::Cubochoric && outputType == OrientationRepresentation::Type::Rodrigues)
   {
     messageHandler(complex::IFilter::Message{complex::IFilter::Message::Type::Info, "Converting Cubochoric to Rodrigues"});
     ConversionFunctionType cu2ro = OrientationTransformation::cu2ro<InputType, OutputType>;
-    parallelAlgorithm.execute(::ConvertOrientation<float, ConversionFunctionType, 3, 4>(inputDataArray, outputDataArray, cu2ro));
+    parallelAlgorithm.execute(::ConvertOrientation<float, ConversionFunctionType, ValidateInputDataFunctionType, 3, 4>(inputDataArray, outputDataArray, cu2ro, cuCheck));
   }
   else if(inputType == OrientationRepresentation::Type::Cubochoric && outputType == OrientationRepresentation::Type::Homochoric)
   {
     messageHandler(complex::IFilter::Message{complex::IFilter::Message::Type::Info, "Converting Cubochoric to Homochoric"});
     ConversionFunctionType cu2ho = OrientationTransformation::cu2ho<InputType, OutputType>;
-    parallelAlgorithm.execute(::ConvertOrientation<float, ConversionFunctionType, 3, 3>(inputDataArray, outputDataArray, cu2ho));
+    parallelAlgorithm.execute(::ConvertOrientation<float, ConversionFunctionType, ValidateInputDataFunctionType, 3, 3>(inputDataArray, outputDataArray, cu2ho, cuCheck));
+  }
+  else if(inputType == OrientationRepresentation::Type::Cubochoric && outputType == OrientationRepresentation::Type::Stereographic)
```

Comment:
There has got to be a better way to strip this boilerplate. I think this is a case where a variadic template with perfect forwarding and/or a generalized policy adapter similar to this: https://stackoverflow.com/questions/7694201/unpacking-parameter-pack-of-args-into-the-constructor-of-each-class-defined-in-a would save a lot of extra code lines and provide much more readability

Reply:
Sure, but this PR is about getting the other 3 filters ported and not fixing ConvertOrientations.